lib/manager/token-manager.js

```diff
@@ -0,0 +1,98 @@
+"use strict";
+
+/**
+ * @typedef {Object} tokenManager~record
+ * @property {string} contentType
+ * @property {string} data
+ * @property {boolean} [public]
+ */
+
+/**
+ * @typedef {Object} tokenManager~createOptions
+ * @property {string} [contentType]
+ * @property {boolean} [public]
+ */
+
+module.exports = (config, promise, random, storageServiceFactory) => {
+    var idLength, storageService;
+
+
+    /**
+     * Creates a new token.
+     *
+     * @param {string} accountId
+     * @param {(Buffer|string)} data
+     * @param {tokenManager~createOptions} options
+     * @return {Promise.<string>} tokenId
+     */
+    function createAsync(accountId, data, options) {
+        var record;
+
+        record = {
+            contentType: "application/octet-stream",
+            data
+        };
+
+        options = options || {};
+
+        if (options.contentType) {
+            record.contentType = options.contentType;
+        }
+
+        if (options.public) {
+            record.public = true;
+        }
+
+        return random.idAsync(idLength).then((tokenId) => {
+            return storageService.putAsync([
```

Comment:
Is it possible for tokens to overwrite another one? I know the hashes are astronomically large, but could it happen?